src/utils/authentication.py

```diff
@@ -6,8 +6,16 @@
 hashed = bcrypt.hashpw(bytes(TOKEN, "utf-8"), bcrypt.gensalt())
 
 
-def verify_password(plain_password):
-    if bcrypt.checkpw(str.encode(plain_password), hashed):
-        return True
-    else:
+def verify_length(token):
+    length = len(token)
+    if length <= 7 or length >= 25:
         return False
+    return True
+
+
+def verify_password(plaintext_token):
```

Comment:
Also, since I can't find the code where the username check is done;
you probably also want to prevent timing attacks there by doing a check similar to pseudocode below:
```
if (exists(username)) {
    verify_password(password);
} else {
    pretend_to_verify_password();
}
```

The reasoning being if you don't do `pretend_to_verify_password` it only calls `verify_password` for an existing/valid username and the call will take *slightly* longer than it would for an invalid username. That in turn allows me to just run a dictionary and see which username is valid.

You can see more info on this [here](https://hexdocs.pm/argon2_elixir/Argon2.html#no_user_verify/1) (it's an Elixir library, but the explanation applies here too)